src/scriptwriter/scenarios/ScenariosHandler.js

```diff
@@ -16,6 +16,7 @@ export default class ScenariosHandler {
             failing: [],
         };
         this._initTime = null;
+        this._index = 0;
```

Comment:
This feels like too much general name for the variable that is used for the whole class, but is meant to be related just to the `_getRandomScenarioStartUrl` method. I would go with `this._startUrlIndex`.

Reply:
Variable renamed to `this._urlPathsIndex`